File: simple-scrape.py
# scraper-python.py
# code by Joe Osborne, https://medium.com/@joerosborne/intro-to-web-scraping-build-your-first-scraper-in-5-minutes-1c36b5c4b110
# To run this script, paste `python scraper-python.py` in the terminal

import requests
from bs4 import BeautifulSoup


def scrape():
    
    url = 'https://www.example.com'
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')
    # print soup: outputs all of the HTML for the page
    # print(soup)
    title = soup.select_one('h1').text
    text = soup.select_one('p').text
    link = soup.select_one('a').get('href')

    print(title)
    print(text)
    print(link)


if __name__ == '__main__':
    scrape()

File: simple-scrape.js
/**
 * scraper-javascript.js
 * code by Joe Osborne, https://medium.com/@joerosborne/intro-to-web-scraping-build-your-first-scraper-in-5-minutes-1c36b5c4b110
 * To run this script, copy and paste `node scraper-javascript.js` in the terminal
 */

const cheerio = require('cheerio');



(async () => {
    const url = 'https://www.example.com';
    const response = await fetch(url);

    const $ = cheerio.load(await response.text());
//   console...html prints the entire page
//   console.log($.html());
    const title = $('h1').text();
    const text = $('p').text();
    const link = $('a').attr('href');

    console.log(title);
    console.log(text);
    console.log(link);

})();